src/js/http_client.js

```diff
@@ -206,6 +220,13 @@ function responseOnEnd() {
   }
 }
 
+ClientRequest.prototype._implicitHeader = function _implicitHeader() {
+  if (this._header) {
+    throw new Error('Cannot render headers after they are sent');
```

Comment:
This code seems say: if header does not exist, it's unable to render again, because it can be sent? Why the `this._header` means the response has been sent?